src/axl_pthread.c

```diff
@@ -65,6 +94,77 @@ unsigned int axl_get_nprocs(void)
     return cpu_threads;
 }
 
+/* Get the pthread data for a given AXL ID */
+struct axl_pthread_data *
+axl_pthread_data_lookup(int id)
+{
+    struct axl_pthread_data *pdata;
+    struct axl_pthread_data *ret = NULL;
+
+    pthread_mutex_lock(&axl_all_pthread_data.lock);
+
+    while ((pdata = axl_all_pthread_data.head)) {
```

Comment:
Do we need a pdata = pdata->next in this loop?

Reply:
Maybe?

pdata = axl_all_pthread_data.head;
while (pdata) {
  ...
  pdata = pdata->next;
}

Reply:
Yep, good catch!